### PR TITLE
Restrict baremetal specific changes to baremetal CIs

### PR DESCRIPTION
**Pull request by fidencio**
#3493 fixed the issues we had with the ARM CI, but the changes done were done for any environment.

Let's restrict those changes a little bit and only run those on baremetal CIs.

Fixes: #3516

### CODE REVIEW

File: integration/containerd/cri/integration-tests.sh [hunk 1]
```diff
@@ -190,7 +190,7 @@ check_daemon_setup() {
 
 	if [ "${BAREMETAL}" == true ]; then
 		#restart docker service as TestImageLoad depends on it
-		restart_docker
+		sudo -E restart_docker
 	fi
```
Comment (jongwu) [inline integration/containerd/cri/integration-tests.sh:194]: Why do we need "sudo" here? And it lead to the failure in arm ci.
`10:49:34 sudo: restart_docker: command not found`

**Comment by jongwu** [comment]
@fidencio - arm ci turns green.

File: integration/kubernetes/init.sh [hunk 2]
```diff
@@ -108,12 +108,12 @@ trap 'sudo -E sh -c "rm -r "${kubeadm_config_file}""' EXIT
 
 if [ "${BAREMETAL}" == true ] && [[ $(wc -l /proc/swaps | awk '{print $1}') -gt 1 ]]; then
```
Comment (jongwu) [inline integration/kubernetes/init.sh:109]: We need let the  "BAREMETAL" check separate from the swap check. So what about:
"
if [[ $(wc -l /proc/swaps | awk '{print $1}') -gt 1 ]]; then
         sudo swapoff -a || true
fi

if [ "${BAREMETAL}" == true &&  "$(command -v kubelet)" != "" ]; then
       #reinstall kubelet to do deep cleanup
      	info "reinstall kubeadm, kubelet before initialize k8s"
	bash -f "${SCRIPT_PATH}/../../.ci/install_kubernetes.sh"	
fi
"

Comment (jongwu) [reply integration/kubernetes/init.sh:109]: The Arm Ci fails as the "wc -l /proc/swaps | awk '{print $1}'" is "1" in the Arm server which results in the "reinstall k8s" ops didn't start.

Comment (fidencio) [reply integration/kubernetes/init.sh:109]: I'm not sure about this, @jongwu.

The baremetal check to disable swap has always been there and shouldn't cause an issue (it wasn't already).
I'll split the check, and let it run a few more times.

Comment (fidencio) [reply integration/kubernetes/init.sh:109]: Ah, now I see, sorry, my bad!

File: integration/kubernetes/init.sh [hunk 3]
```diff
@@ -111,7 +111,7 @@ if [ "${BAREMETAL}" == true ] && [[ $(wc -l /proc/swaps | awk '{print $1}') -gt
 fi
 
 #reinstall kubelet to do deep cleanup
-if [ "$(command -v kubelet)" != "" ]; then
+if [ "{BAREMETAL}" == true && "$(command -v kubelet)" != "" ]; then
```
Comment (jongwu) [inline integration/kubernetes/init.sh:114]: Hi @fidencio -, Try `if [ "${BAREMETAL}" == true -a "$(command -v kubelet)" != "" ]; then`

File: integration/kubernetes/init.sh [hunk 4]
```diff
@@ -111,7 +111,7 @@ if [ "${BAREMETAL}" == true ] && [[ $(wc -l /proc/swaps | awk '{print $1}') -gt
 fi
 
 #reinstall kubelet to do deep cleanup
-if [ "$(command -v kubelet)" != "" ]; then
+if [ "{BAREMETAL}" == true -a "$(command -v kubelet)" != "" ]; then
```
Comment (jongwu) [inline integration/kubernetes/init.sh:114]: miss "$" before "{BAREMETAL}"

Comment (fidencio) [reply integration/kubernetes/init.sh]: Lack of attention is  killing me. :-)

**Comment by fidencio** [comment]
Okay, it passes, now let's give it a try on a second run.

**Comment by fidencio** [comment]
@jongwu, now it's good to go! :-)

**Review by jcvenegas — APPROVED** [review]

**Comment by fidencio** [comment]
@snir911, may I bother you for an ack?

**Review by snir911 — APPROVED** [review]
LGTM!

**Pull request merged**